test/org/sagebionetworks/bridge/services/HealthCodeServiceImplTest.java

```diff
@@ -41,19 +45,60 @@ public void after() {
     }
 
     @Test
-    public void test() {
+    public void createAndRetrieveHealthId() {
         Study study = studyService.getStudy(TEST_STUDY_IDENTIFIER);
-        HealthId healthId1 = healthCodeService.create(study);
-        assertNotNull(healthId1);
-        assertEquals(healthId1.getCode(), healthCodeService.getHealthCode(healthId1.getId()));
-        HealthId healthId2 = healthCodeService.create(study);
-        assertFalse(healthId1.getId().equals(healthId2.getId()));
-        assertFalse(healthId1.getCode().equals(healthId2.getCode()));
-        assertEquals(TEST_STUDY_IDENTIFIER, healthCodeService.getStudyIdentifier(healthId1.getCode()));
-        assertEquals(TEST_STUDY_IDENTIFIER, healthCodeService.getStudyIdentifier(healthId2.getCode()));
-        assertNull(healthCodeService.getStudyIdentifier("abcdef"));
+        HealthId mapping = healthCodeService.createMapping(study);
+        assertNotNull(mapping);
+        assertEquals(mapping.getCode(), healthCodeService.getMapping(mapping.getId()).getCode());
+        HealthId healthId2 = healthCodeService.createMapping(study);
+        assertFalse(mapping.getId().equals(healthId2.getId()));
+        assertFalse(mapping.getCode().equals(healthId2.getCode()));
     }
+    
+    @Test
+    public void invalidHealthIdReturnsNull() {
+        HealthIdDao dao = mock(HealthIdDao.class);
+        HealthCodeServiceImpl healthCodeService = new HealthCodeServiceImpl();
+        healthCodeService.setHealthIdDao(dao);
 
+        HealthId healthId = healthCodeService.getMapping(null);
+        assertNull(healthId);
+        
+        verifyZeroInteractions(dao);
+    }
+    
+    @Test
+    public void invalidHealthIdReturnsNullNotInvalidHealthIdObject() {
+        HealthIdDao dao = mock(HealthIdDao.class);
+        when(dao.getCode("123")).thenReturn("abc");
+        when(dao.getCode("456")).thenReturn(null);
+        
+        HealthCodeServiceImpl healthCodeService = new HealthCodeServiceImpl();
+        healthCodeService.setHealthIdDao(dao);
+        
+        // valid
+        HealthId id1 = healthCodeService.getMapping("123");
+        assertEquals("123", id1.getId());
+        assertEquals("abc", id1.getCode());
+        
+        // invalid... should return no object at all.
+        HealthId id2 = healthCodeService.getMapping("456");
+        assertNull(id2);
+        
+    }
+    
+    /*
+   @Override
+    public HealthId getMapping(String healthId) {
+        checkNotNull(healthId);
+        final String healthCode = healthIdDao.getCode(healthId);
+        if (healthCode == null) {
```

Comment:
[MINOR] Should probably remove this commented out implementation of getMapping, especially since it's already diverged from the actual implementation.